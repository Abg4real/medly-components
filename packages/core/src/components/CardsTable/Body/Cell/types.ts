import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
