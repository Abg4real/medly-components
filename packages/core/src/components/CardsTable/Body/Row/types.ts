import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    withWhiteBackground: boolean;
}
