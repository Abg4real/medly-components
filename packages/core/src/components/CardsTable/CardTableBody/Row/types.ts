import { Props as CardProps } from '../../../Card/types';

export interface Props extends CardProps {
    disabled?: boolean;
    withLightTheme: boolean;
}
