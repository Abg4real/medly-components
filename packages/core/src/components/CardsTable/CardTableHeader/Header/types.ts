import { SortOrder } from '../../types';
import { WithThemeProp } from '@medly-components/utils/src';

export interface Props extends WithThemeProp{
    sortField?: string;
    field?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
    enableSorting?: boolean;
    title: string;
}    