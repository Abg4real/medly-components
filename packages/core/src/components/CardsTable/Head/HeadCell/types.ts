import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
