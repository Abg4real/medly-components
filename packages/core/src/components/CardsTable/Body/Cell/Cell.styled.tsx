import { styled } from '@medly-components/utils';
import Card from '../../../Card';

export const CellContent = styled(Card)`
    padding: 15px 20px;
`;
