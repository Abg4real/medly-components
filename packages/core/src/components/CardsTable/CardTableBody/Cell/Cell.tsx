import React, { useCallback, useRef } from 'react';
import Text from '../../../Text';
import { Props } from './types';


export const Cell: React.SFC<Props> = React.memo(props => {
    const childRef = useRef(null),
        {
            data,
            rowId,
            column,
            isRowClickDisabled,
        } = props
    const formattedCell = useCallback(() => {
            switch (column.formatter) {
                case 'boolean':
                    return <Text> {data} </Text>;
                case 'react-component': {
                    const Component = column.component;
                    return (
                        <div ref={childRef}>
                            <Component {...{ data, rowId, disabled: isRowClickDisabled }} />
                        </div>
                    );
                }
                default:
                    return <Text>{data}</Text>;
            }
        }, [data, rowId, isRowClickDisabled, column]);


    return (
        <>
            {formattedCell()}
        </>
    );
});
Cell.displayName = 'Cell';
