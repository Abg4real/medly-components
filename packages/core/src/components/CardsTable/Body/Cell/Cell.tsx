import React, { useCallback } from 'react';
import Text from '../../../Text';
import { Props } from './types';
import * as Style from './Cell.styled';
import { WithStyle } from '@medly-components/utils/src';



export const Cell: React.SFC<Props> & WithStyle = React.memo(props => {
    const {
            data,
            rowId,
            column,
            isRowClickDisabled,
        } = props
    const formattedCell = useCallback(() => {
            switch (column.formatter) {
                case 'boolean':
                    return <Text> {data} </Text>;
                case 'react-component': {
                    const Component = column.component;
                    return (
                        <Component {...{ data, rowId, disabled: isRowClickDisabled }} />
                    );
                }
                default:
                    return <Text>{data}</Text>;
            }
        }, [data, rowId, isRowClickDisabled, column]);


    return (
        <Style.CellContent variant="flat" flowDirection="vertical" alignItems="left" >
            {formattedCell()}
        </Style.CellContent>
    );
});
Cell.displayName = 'Cell';
