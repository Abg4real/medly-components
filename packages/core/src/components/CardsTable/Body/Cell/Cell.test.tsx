import React from 'react';
import Text from '../../../Text';
import { ColumnConfig } from '../../types';
import { Cell } from './Cell';
import { render } from '@testing-library/react';


describe('Cell', () => {
    const columnConfig: ColumnConfig = {
        title: 'Column 1',
        field: 'column1',
    };
    
    it('should render cell with text properly', () => {
        const { container } = render(
            <Cell column={columnConfig} data="text" rowId="1" />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render cell with component properly', () => {
        const config: ColumnConfig = {
            ...columnConfig,
            component: Text
        }
        const { container } = render(
            <Cell column={config} data="text" rowId="1" />
        );
        expect(container).toMatchSnapshot();
    });

});
