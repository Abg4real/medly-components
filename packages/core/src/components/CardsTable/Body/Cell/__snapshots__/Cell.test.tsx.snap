// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should render text properly 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
}

.c0.c0 * {
  background-color: transparent;
}

.c1 {
  padding: 15px 20px;
}

<div>
  <div
    class="c0 c1"
  >
    <span
      class="c2"
    >
       
      text
       
    </span>
  </div>
</div>
`;
